bootstrap/bootstrap.go

```diff
@@ -390,29 +390,29 @@ func (b *Bootstrap) executeHook(ctx context.Context, scope string, name string,
 	}
 
 	// Finally, apply changes to the current shell and config
-	b.applyEnvironmentChanges(changes.Env, changes.Dir, redactors)
+	b.applyEnvironmentChanges(changes, redactors)
 	return nil
 }
 
-func (b *Bootstrap) applyEnvironmentChanges(environ *env.Environment, dir string, redactors RedactorMux) {
-	if dir != b.shell.Getwd() {
-		_ = b.shell.Chdir(dir)
+func (b *Bootstrap) applyEnvironmentChanges(changes hook.HookScriptChanges, redactors RedactorMux) {
+	if changes.Dir != b.shell.Getwd() {
+		_ = b.shell.Chdir(changes.Dir)
 	}
 
 	// Do we even have any environment variables to change?
-	if environ == nil || environ.Length() == 0 {
+	if changes.Diff.Empty() {
 		return
 	}
 
-	mergedEnv := b.shell.Env.Merge(environ)
+	mergedEnv := b.shell.Env.Apply(changes.Diff)
 
 	// reset output redactors based on new environment variable values
 	redactors.Flush()
 	redactors.Reset(getValuesToRedact(b.shell, b.Config.RedactedVars, mergedEnv.ToMap()))
 
 	// First, let see any of the environment variables are supposed
 	// to change the bootstrap configuration at run time.
-	bootstrapConfigEnvChanges := b.Config.ReadFromEnvironment(environ)
+	bootstrapConfigEnvChanges := b.Config.ReadFromEnvironment(mergedEnv)
```

Comment:
We want to double check this part, previously we passed `environ` which was just the _diff_ not the merged environment. Will this print report the config variables as being changed now?